src/main/java/io/confluent/castle/action/SchemaRegistryStartAction.java

```diff
@@ -0,0 +1,249 @@
+            osw.write(String.format("log4j.appender.kafkaAppender.DatePattern='.'yyyy-MM-dd-HH%n"));
+            osw.write(String.format("log4j.appender.kafkaAppender.File=%s/server.log%n", SCHEMA_REGISTRY_LOGS));
+            osw.write(String.format("log4j.appender.kafkaAppender.layout=org.apache.log4j.PatternLayout%n"));
+            osw.write(String.format("log4j.appender.kafkaAppender.layout.ConversionPattern=%s%n%n",
+                "[%d] %p %m (%c)%n"));
+            osw.write(String.format("log4j.logger.kafka=ERROR%n"));
+            osw.write(String.format("log4j.logger.org.apache.kafka=ERROR%n"));
+            osw.write(String.format("log4j.logger.org.apache.zookeeper=ERROR%n"));
+            osw.write(String.format("log4j.logger.org.I0Itec.zkclient.ZkClient=INFO%n"));
+            osw.write(String.format("log4j.additivity.kafka.server=false%n"));
+            osw.write(String.format("log4j.additivity.kafka.consumer.ZookeeperConsumerConnector=false%n"));
+            success = true;
+            return file;
+        } finally {
+            CastleUtil.closeQuietly(cluster.clusterLog(),
+                osw, "temporary Schema Registry file OutputStreamWriter");
+            CastleUtil.closeQuietly(cluster.clusterLog(),
+                fos, "temporary Schema Registry file FileOutputStream");
+            if (!success) {
+                CastleUtil.deleteFileOrLog(node.log(), file);
+            }
+        }
+    }
+
+    private void writeSchemas(final CastleCluster cluster, final CastleNode node) throws Throwable {
+        for (int i = 0; i < role.schemas().size(); i++) {
+            File schemaFile = writeSchema(cluster, node, i);
+            node.uplink().command().syncTo(schemaFile.getAbsolutePath(),
+                String.format(ActionPaths.SCHEMA_REGISTRY_SCHEMA, i)).mustRun();
+        }
+    }
+
+    private File writeSchema(CastleCluster cluster, CastleNode node, int index) throws IOException {
+        File file = null;
+        FileOutputStream fos = null;
+        OutputStreamWriter osw = null;
+        boolean success = false;
+        try {
+            Schema schema = role.schemas().get(index);
+            file = new File(cluster.env().workingDirectory(),
+                    String.format("schema-%d.asc", index));
+            fos = new FileOutputStream(file, false);
+            osw = new OutputStreamWriter(fos, StandardCharsets.UTF_8);
+            osw.write("{ \"schema\": \"");
+            osw.write(schema.schema().replace("\"", "\\\""));
+            osw.write("\" }");
+            success = true;
+            return file;
+        } finally {
+            CastleUtil.closeQuietly(cluster.clusterLog(),
+                osw, "temporary Schema Registry file OutputStreamWriter");
+            CastleUtil.closeQuietly(cluster.clusterLog(),
+                fos, "temporary Schema Registry file FileOutputStream");
+            if (!success) {
```

Comment:
Why delete logs upon failure? I thought we want to leave it for debugging still?

Reply:
These files are copied from the castle client to the EC2 instance.  It seems the pattern throughout `castle` is to delete the files in EC2 in case of failure.  I think they can still be inspected on the client.

Reply:
SG.